src/code42cli/cmds/securitydata.py

```diff
@@ -358,23 +365,77 @@ def search(
     include_all,
     **kwargs,
 ):
+
     """Search for file events."""
-    output_header = ext.try_get_default_header(
+    output_header = try_get_default_header(
         include_all, _create_search_header_map(), format
     )
     formatter = FileEventsOutputFormatter(format, output_header)
     cursor = _get_cursor(state, use_checkpoint)
-    handlers = ext.create_handlers(
-        state.sdk,
-        FileEventExtractor,
-        cursor,
-        use_checkpoint,
-        formatter=formatter,
-        force_pager=include_all,
-    )
-    _extract(
-        state, handlers, begin, end, or_query, advanced_query, saved_search, **kwargs
-    )
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        # if checkpoint name exists, checkpoint should be that eventId,
+        # otherwise it should set the initial value to ""
+        checkpoint = cursor.get(checkpoint_name) or ""
+
+        # older app versions stored checkpoint as float timestamp.
+        # we handle those here until the next run containing events will store checkpoint as the last eventId
+        if isinstance(checkpoint, (int, float)):
```

Comment:
Hrm, since the FileEventCursorStore.get() returns a string, this will never be true... We'll have to try calling `float()` on the checkpoint instead of `isinstance()`.